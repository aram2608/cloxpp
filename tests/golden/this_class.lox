class Counter {
  init(value) {
    this.value = value;
  }

  increment() {
    this.value = this.value + 1;
  }
}

var c = Counter(5);
c.increment();
print c.value; // expect 6