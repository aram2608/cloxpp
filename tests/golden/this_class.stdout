Token: CLASS class 1
Token: LEFT_BRACE { 1
Token: IDENTIFIER init 2
Token: LEFT_PAREN ( 2
Token: IDENTIFIER value 2
Token: RIGHT_PAREN ) 2
Token: LEFT_BRACE { 2
Token: THIS this 3
Token: DOT . 3
Token: IDENTIFIER value 3
Token: EQUAL = 3
Token: IDENTIFIER value 3
Token: SEMICOLON ; 3
Token: RIGHT_BRACE } 4
Token: IDENTIFIER increment 6
Token: LEFT_PAREN ( 6
Token: RIGHT_PAREN ) 6
Token: LEFT_BRACE { 6
Token: THIS this 7
Token: DOT . 7
Token: IDENTIFIER value 7
Token: EQUAL = 7
Token: THIS this 7
Token: DOT . 7
Token: IDENTIFIER value 7
Token: PLUS + 7
Token: NUMBER 1 7
Token: SEMICOLON ; 7
Token: RIGHT_BRACE } 8
Token: RIGHT_BRACE } 9
Token: VAR var 11
Token: IDENTIFIER c 11
Token: EQUAL = 11
Token: LEFT_PAREN ( 11
Token: NUMBER 5 11
Token: RIGHT_PAREN ) 11
Token: SEMICOLON ; 11
Token: IDENTIFIER c 12
Token: DOT . 12
Token: IDENTIFIER increment 12
Token: LEFT_PAREN ( 12
Token: RIGHT_PAREN ) 12
Token: SEMICOLON ; 12
Token: PRINT print 13
Token: IDENTIFIER c 13
Token: DOT . 13
Token: IDENTIFIER value 13
Token: SEMICOLON ; 13
Token: EOF  13
